### ISSUE TITLE
Darktable doesn't launch on M1 Max Mac

### ISSUE THREAD

**Issue by JesseRodriguez** [darktable-org/darktable#16926]
### Describe the bug

I downloaded Darktable using various methods (following instructions to disable the gatekeeper), and it always gets to the point where when I attempt to run the application, literally nothing happens, just seems to immediately exit. 

### Steps to reproduce

Simplest path is:
1. `brew install --cask --no-quarantine darktable`
2. control-click + open in Finder `/Applications` folder

### Expected behavior

darktable should launch

### Logfile | Screenshot | Screencast

'''2024-06-02 22:23:42.764592 (gui/501/application.org.darktable.125519130.125519143) <Notice>: internal event: WILL_SPAWN, code = 0
2024-06-02 22:23:42.764601 (gui/501/application.org.darktable.125519130.125519143) <Notice>: service state: spawn scheduled
2024-06-02 22:23:42.764602 (gui/501/application.org.darktable.125519130.125519143) <Notice>: service state: spawning
2024-06-02 22:23:42.764612 <Notice>: Coalition Cache Hit: app<application.org.darktable.125519130.125519143(501)> [15096]
2024-06-02 22:23:42.764624 (gui/501/application.org.darktable.125519130.125519143) <Notice>: launching: launch job demand
2024-06-02 22:23:42.765041 (gui/501/application.org.darktable.125519130.125519143 [34042]) <Notice>: xpcproxy spawned with pid 34042
2024-06-02 22:23:42.765055 (gui/501/application.org.darktable.125519130.125519143 [34042]) <Notice>: internal event: SPAWNED, code = 0
2024-06-02 22:23:42.765056 (gui/501/application.org.darktable.125519130.125519143 [34042]) <Notice>: service state: xpcproxy
2024-06-02 22:23:42.765080 (gui/501/application.org.darktable.125519130.125519143 [34042]) <Notice>: internal event: SOURCE_ATTACH, code = 0
2024-06-02 22:23:42.773810 (gui/501/application.org.darktable.125519130.125519143 [34042]) <Notice>: service state: running
2024-06-02 22:23:42.773817 (gui/501/application.org.darktable.125519130.125519143 [34042]) <Notice>: internal event: INIT, code = 0
2024-06-02 22:23:42.773819 (gui/501/application.org.darktable.125519130.125519143 [34042]) <Notice>: job state = running
2024-06-02 22:23:42.773839 (gui/501/application.org.darktable.125519130.125519143 [34042]) <Notice>: Successfully spawned darktable[34042] because launch job demand
2024-06-02 22:23:42.825737 (pid/34042 [darktable]) <Notice>: uncorking exec source upfront
2024-06-02 22:23:42.825753 (pid/34042 [darktable]) <Notice>: created
2024-06-02 22:23:42.842672 (gui/501/application.org.darktable.125519130.125519143 [34042]) <Notice>: exited due to exit(1), ran for 77ms
2024-06-02 22:23:42.842681 (gui/501/application.org.darktable.125519130.125519143 [34042]) <Notice>: service state: exited
2024-06-02 22:23:42.842687 (gui/501/application.org.darktable.125519130.125519143 [34042]) <Notice>: internal event: EXITED, code = 0
2024-06-02 22:23:42.842690 (gui/501/application.org.darktable.125519130.125519143 [34042]) <Notice>: job state = exited
2024-06-02 22:23:42.842717 (gui/501 [100011]) <Notice>: service inactive: application.org.darktable.125519130.125519143
2024-06-02 22:23:42.842720 (gui/501/application.org.darktable.125519130.125519143 [34042]) <Notice>: service state: not running
2024-06-02 22:23:42.842726 (pid/34042 [darktable]) <Notice>: shutting down
2024-06-02 22:23:42.842731 (pid/34042 [darktable]) <Notice>: cleaning up
2024-06-02 22:23:42.842736 (system) <Notice>: removing child: pid/34042
2024-06-02 22:23:42.843921 (gui/501/application.org.darktable.125519130.125519143) <Notice>: removing job: caller = runningboardd
2024-06-02 22:23:42.843930 (gui/501 [100011]) <Notice>: removing service: application.org.darktable.125519130.125519143
2024-06-02 22:23:42.843938 (gui/501/application.org.darktable.125519130.125519143) <Notice>: internal event: PETRIFIED, code = 0
2024-06-02 22:23:42.843940 (gui/501/application.org.darktable.125519130.125519143) <Notice>: job state = removed'''

### Where did you obtain darktable from?

distro packaging

### darktable version

4.6.1

### What OS are you using?

Mac

### What is the version of your OS?

Sonoma 14.5

### Describe your system?

M1 Max Mac (2021), 64 GB RAM

### Are you using OpenCL GPU in darktable?

I dont know

### If yes, what is the GPU card and driver?

n/a

### Please provide additional context if applicable. You can attach files too, but might need to rename to .txt or .zip

n/a

**Comment by zisoft** [comment]
Can you please try to download darktable from the [Release page](https://github.com/darktable-org/darktable/releases/tag/release-4.6.1) (scroll down to the assets section).

Then follow the [Installation notes](https://www.darktable.org/install/#macos) before the first launch.

**Comment by JesseRodriguez** [comment]
Yes, sorry if I wasn't clear; I have tried the installation by downloading `darktable-4.6.1-arm64.dmg` and then either running the xattr command either before mounting or after installation, both methods result in the behavior described above. I just noted the homebrew install because it was the quickest way. I've also installed via homebrew without the --no-quarantine flag and run the xattr command afterwards, same result. Tried restarting, updating MacOS, all to no avail.

**Comment by MStraeten** [comment]
Please run darktable from terminal via `/Applications/darktable.app/Contents/MacOS/darktable -d common > ~/darktable.log ` and then upload the generated logfile.
In general darktable runs fine on a m1 max

**Comment by JesseRodriguez** [comment]
Ok I got it to run in the following way: first time trying to run from terminal, got 

`path lookup '/Users/jesse/.config/darktable' fails with: 'No such file or directory'`

So I 

`sudo mkdir -p ~/.config/darktable`

and ran again. Got a database locking error, which from the logfile looked like it was coming from a failure to backup or read in the `.config` file, so then I went

```
sudo chown -R $USER: $HOME/.config/darktable
chmod -R u+rw $HOME/.config/darktable
```

And now it seems to be running. Did some test edits on a couple RAW files and exported fine. I'll close the issue. Thanks for the terminal command which redirects the debug info to a logfile, didn't know how to do that before!
